src/main/java/com/api/util/ApiSecurity/ApiSigning.java

```diff
@@ -411,16 +408,21 @@ public static String getBaseString(String authPrefix
             ApiList paramList = new ApiList();
 
             // process QueryString from url by transfering it to paramList
-            if (siteUri.getQuery().length() > 1) {
+            if (null != siteUri.getQuery()) {
                 String queryString = siteUri.getRawQuery();
                 log.debug("queryString:: {}", queryString);
 
                 String[] paramArr = queryString.split("&");
                 for (String item : paramArr) {
-                    log.debug("item:: {}", item);
+                    log.debug("queryItem:: {}", item);
                     String[] itemArr = item.split("=");
                     try {
-                        paramList.add(itemArr[0], java.net.URLDecoder.decode(itemArr[1], StandardCharsets.UTF_8.toString()));
+                    	if(itemArr.length == 1) {
+                    		paramList.add(itemArr[0], "");
```

Comment:
It will be passing as **key = ''** as an empty string for the basestring signing?

Reply:
The BaseString will be resolve to: http://localhost/api/resource?key&key2=123

So , equal sign will be dropped